chancloser.go

```diff
@@ -263,6 +268,23 @@ func (c *channelCloser) CloseRequest() *htlcswitch.ChanClose {
 	return c.closeReq
 }
 
+// checkRemoteDeliveryAddress checks that the address provided by the remote
+// peer matches the upfront shutdown address, if it is set. It does not perform
+// any validation on the address provided.
+func checkRemoteDeliveryAddress(upfront, address lnwire.DeliveryAddress) error {
+	// If an upfront shutdown address is empty, return early.
+	if len(upfront) == 0 {
```

Comment:
So here we are checking whether the address our peer has provided matches the pre-negotiated upfront address. There are two cases:
1. They did not negotiate an upfront address (it's an optional feature), so we do not need to enforce close out to a specific address
2. They did negotiate an upfront address, and we need to check that they're sticking to it; if not, we need to disconnect as per the spec's instruction because they may be comporomised